mmdet/models/losses/gfocal_loss.py

```diff
@@ -0,0 +1,107 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+from .utils import weight_reduce_loss
+
+
+def quality_focal_loss(pred,
+                       label,
+                       score,
+                       weight=None,
+                       beta=2.0,
+                       reduction='mean',
+                       avg_factor=None):
+    # all goes to 0
+    pred_sigmoid = pred.sigmoid()
+    pt = pred_sigmoid
+    zerolabel = pt.new_zeros(pred.shape)
+    loss = F.binary_cross_entropy_with_logits(
+        pred, zerolabel, reduction='none') * pt.pow(beta)  # pt = pt.abs()
+    # find positive positions and quality labels
+    label = label - 1
+    pos = (label >= 0).nonzero().squeeze(1)
+    a = pos
+    b = label[pos].long()
+    # positive goes to bbox quality, e.g., IoU target
+    pt = score[a] - pred_sigmoid[a, b]
+    loss[a, b] = F.binary_cross_entropy_with_logits(
+        pred[a, b], score[a], reduction='none') * pt.abs().pow(beta)
+    loss = weight_reduce_loss(loss, weight, reduction, avg_factor)
+    return loss
+
+
+def distribution_focal_loss(pred,
+                            label,
+                            weight=None,
+                            reduction='mean',
+                            avg_factor=None):
+    disl = label.long()
```

Comment:
1. disl -> long_label
2. change `disr` to a more clear name

Reply:
Got it, one more question, should I close this pull request first, then update related code to perform a new pull request?